flink-end-to-end-tests/run-nightly-tests.sh

```diff
@@ -58,25 +58,97 @@ fi
-  $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 2 4
+  STATE_BACKEND_TYPE=file STATE_BACKEND_FILE_ASYNC=false $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 2 2
   EXIT_CODE=$?
 fi
 
 if [ $EXIT_CODE == 0 ]; then
   printf "\n==============================================================================\n"
-  printf "Running Resuming Savepoint (scale down) end-to-end test\n"
+  printf "Running Resuming Savepoint (file, async, scale up) end-to-end test\n"
   printf "==============================================================================\n"
-  $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 4 2
+  STATE_BACKEND_TYPE=file STATE_BACKEND_FILE_ASYNC=true $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 2 4
+  EXIT_CODE=$?
+fi
+
+if [ $EXIT_CODE == 0 ]; then
+  printf "\n==============================================================================\n"
+  printf "Running Resuming Savepoint (file, sync, scale up) end-to-end test\n"
+  printf "==============================================================================\n"
+  STATE_BACKEND_TYPE=file STATE_BACKEND_FILE_ASYNC=false $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 2 4
+  EXIT_CODE=$?
+fi
+
+if [ $EXIT_CODE == 0 ]; then
+  printf "\n==============================================================================\n"
+  printf "Running Resuming Savepoint (file, async, scale down) end-to-end test\n"
+  printf "==============================================================================\n"
+  STATE_BACKEND_TYPE=file STATE_BACKEND_FILE_ASYNC=true $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 4 2
+  EXIT_CODE=$?
+fi
+
+if [ $EXIT_CODE == 0 ]; then
+  printf "\n==============================================================================\n"
+  printf "Running Resuming Savepoint (file, sync, scale down) end-to-end test\n"
+  printf "==============================================================================\n"
+  STATE_BACKEND_TYPE=file STATE_BACKEND_FILE_ASYNC=false $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 4 2
+  EXIT_CODE=$?
+fi
+
+if [ $EXIT_CODE == 0 ]; then
+  printf "\n==============================================================================\n"
+  printf "Running Resuming Savepoint (rocks, non-incremental, no parallelism change) end-to-end test\n"
+  printf "==============================================================================\n"
+  STATE_BACKEND_TYPE=rocks STATE_BACKEND_ROCKS_INCREMENTAL=false $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 2 2
+  EXIT_CODE=$?
+fi
+
+if [ $EXIT_CODE == 0 ]; then
+  printf "\n==============================================================================\n"
+  printf "Running Resuming Savepoint (rocks, incremental, no parallelism change) end-to-end test\n"
+  printf "==============================================================================\n"
+  STATE_BACKEND_TYPE=rocks STATE_BACKEND_ROCKS_INCREMENTAL=true $END_TO_END_DIR/test-scripts/test_resume_savepoint.sh 2 2
```

Comment:
Taking savepoints is not different between incremental and non-incremental RocksDB, so all those incremental tests do essentially the same. While it does not hurt to test them in a strict sense, we might save some time by dropping the incremental case. Or maybe we can keep just one (incremental, no resscale) and drop all incremental rescalings because they currently add nothing to the test coverage.

Reply:
This makes sense, will address this.